mucommander-protocol-gcs/src/main/java/com/mucommander/commons/file/protocol/gcs/GoogleCloudStorageBucket.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * This file is part of muCommander, http://www.mucommander.com
+ * <p>
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * <p>
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.mucommander.commons.file.protocol.gcs;
+
+import com.google.cloud.storage.*;
+import com.mucommander.commons.file.FileURL;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Representation of the Bucket as a Folder for the CloudStorage. The bucket lists its content as its children.
+ *
+ * @author miroslav.spak
+ */
+public class GoogleCloudStorageBucket extends GoogleCloudStorageAbstractFile {
+
+    protected static final Pattern BUCKER_NAME_BLOB_PATH_PATTERN = Pattern.compile("^/([^/]+)/?(.*)/?$");
+
+    private Bucket bucket;
+
+    GoogleCloudStorageBucket(FileURL url) {
+        super(url);
+    }
+
+    GoogleCloudStorageBucket(FileURL url, Bucket bucket) {
+        this(url);
+        this.bucket = bucket;
+    }
+
+    /**
+     * Reads bucket name from the standard {@link FileURL} representation. I.e., first level after the root separator.
+     */
+    protected String getBucketName() {
+        // Find the first part of the path that represents bucket name
+        var matcher = BUCKER_NAME_BLOB_PATH_PATTERN.matcher(fileURL.getPath());
+        return Optional.of(matcher).filter(Matcher::find).map(match -> match.group(1)).orElse("");
+    }
+
+    protected static String getBlobName(BlobInfo blob) {
+        var blobPath = blob != null ? blob.getName() : "";
+        // Find the last part of the path that represents blob file name
+        return new File(blobPath).getName();
```

Comment:
I was looking for the easiest way to get the filename from the path, and this is what Google came out with. I ran it on Windows and Mac, and it worked on both.
But on the second look, I found `FileURL.getFilenameFromPath`, so it might be better.

Reply:
Ok, I also need the names of directories, which is a problem with the method above once the name ends with the separator.
And because the names of blobs look like relative paths, FileURL has a problem with that too.
So the construct above is the easiest way that is working.